### ISSUE TITLE
[BUG] Issues coming from commented code

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#361]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

0.3.0-alpha.5

### Priority

Minor

### Current Behavior

Code inside of comments is being also analyzed. See for instance issue "Searches for implementation class names via properties have been dropped", with rule ID `eap8-xml-binding-00005`.
![image](https://github.com/konveyor/analyzer-lsp/assets/4527008/567792de-b9b6-419c-bbe8-af2bc6f4ff17)

### Expected Behavior

Code within comments should be ignored

### How Reproducible

Always (Default)

### Steps To Reproduce

1. With the given environment and configuration, analyze the daytrader app.
2. Find for instance the "Searches for implementation class names via properties have been dropped" issue.
3. Go to "View affected files"
4. Click on the file `/root/.m2/repository/javax/javaee-api/7.0/javax/xml/bind/JAXBContext.java` and see the incidents.

### Environment

```markdown
* Fedora 38
* minikube version: v1.31.2
* Konveyor 0.3.0-alpha.5
* [sample.daytrader7 app (master branch)](https://github.com/WASdev/sample.daytrader7)
* Analysis configuration:
** Source code + dependencies
** eap8, quarkus, openjdk17, azure, cloud-readiness
** Application and internal dependencies only
** No custom rules
** No additional advanced options
```
